src/components/asthma/components/AsthmaAlertTakeoverNotice/AsthmaAlertTakeoverNotice.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import { Button, LoadingIndicator } from '../../../presentational';
+import './AsthmaAlertTakeoverNotice.css';
+import MyDataHelps from '@careevolution/mydatahelps-js';
+import { useInitializeView } from '../../../../helpers/Initialization';
+import { asthmaDataService, dateToAsthmaLogEntryIdentifier } from '../../helpers';
+import { add } from 'date-fns';
+import language from '../../../../helpers/language';
+
+export interface AsthmaAlertTakeoverNoticeProps {
+    previewState?: 'loading' | 'loaded';
+    message: string;
+    logEntrySurveyName: string;
+    innerRef?: React.Ref<HTMLDivElement>;
+}
+
+export default function (props: AsthmaAlertTakeoverNoticeProps) {
+    const [loading, setLoading] = useState<boolean>(true);
+
+    useInitializeView(() => {
+        setLoading(true);
+
+        if (props.previewState) {
+            setLoading(props.previewState === 'loading');
+            return;
+        }
+
+        asthmaDataService.loadLogEntries(add(new Date(), {days: -2})).then(entries => {
+            const todayLogEntry = entries.find(e => e.identifier === dateToAsthmaLogEntryIdentifier(new Date()));
```

Comment:
dateToAsthmaLogEntryIdentifier can be called just once vs and compared on each iteration vs called + compared

Reply:
Done.